lib/src/main/java/com/auth0/android/lock/AuthenticationCallback.java

```diff
@@ -42,32 +40,24 @@
  */
 public abstract class AuthenticationCallback implements LockCallback {
 
-    private static final String TAG = AuthenticationCallback.class.getSimpleName();
-
     /**
      * Called when the authentication flow finished successfully.
      *
      * @param credentials with the tokens.
      */
     public abstract void onAuthentication(@NonNull Credentials credentials);
 
-    /**
-     * Called when the user goes back and closes the activity, without using an Authentication flow.
-     */
-    public abstract void onCanceled();
-
     @Override
     public void onEvent(@LockEvent int event, @NonNull Intent data) {
-        switch (event) {
-            case LockEvent.CANCELED:
-                onCanceled();
-                break;
-            case LockEvent.AUTHENTICATION:
-                parseAuthentication(data);
-                break;
-            case LockEvent.RESET_PASSWORD:
-            case LockEvent.SIGN_UP:
-                break;
+        if (data.hasExtra(Constants.EXCEPTION_EXTRA)) {
+            onError((AuthenticationException) data.getSerializableExtra(Constants.EXCEPTION_EXTRA));
+            return;
+        }
+        if (event == LockEvent.AUTHENTICATION) {
+            Credentials credentials = extractCredentials(data);
+            onAuthentication(credentials);
+        } else if (event == LockEvent.CANCELED) {
+            onError(new AuthenticationException("a0.authentication_canceled", "The user pressed back"));
```

Comment:
If the event is "canceled", then raise an exception with the specific code that represents this. This is only for `AuthenticationException`. `LockCallback` would still receive this event and is up to the dev to handle (this advanced scenario remains available)